editor/connections_dialog.cpp

```diff
@@ -1339,35 +1313,15 @@ void ConnectionsDock::update_tree() {
 	Ref<Script> script_base = selected_node->get_script();
 
 	while (native_base != StringName()) {
-		String class_name;
 		String doc_class_name;
-		String class_brief;
 		Ref<Texture2D> class_icon;
 		List<MethodInfo> class_signals;
 
 		if (script_base.is_valid()) {
-			class_name = script_base->get_global_name();
-			if (class_name.is_empty()) {
-				class_name = script_base->get_path().get_file();
-			}
-
 			doc_class_name = script_base->get_global_name();
 			if (doc_class_name.is_empty()) {
 				doc_class_name = script_base->get_path().trim_prefix("res://").quote();
-			}
-
-			// For a script class, the cache is filled each time.
-			if (!doc_class_name.is_empty()) {
-				if (descr_cache.has(doc_class_name)) {
-					descr_cache[doc_class_name].clear();
-				}
-				HashMap<String, DocData::ClassDoc>::ConstIterator F = doc_data->class_list.find(doc_class_name);
-				if (F) {
-					class_brief = F->value.brief_description;
-					for (int i = 0; i < F->value.signals.size(); i++) {
-						descr_cache[doc_class_name][F->value.signals[i].name] = F->value.signals[i].description;
-					}
-				} else {
+				if (!doc_data->class_list.find(doc_class_name)) {
```

Comment:
This is still not equivalent. The old logic was (simplified):

```cpp

	doc_class_name = script_base->get_global_name();
	if (doc_class_name.is_empty()) {
		doc_class_name = script_base->get_path().trim_prefix("res://").quote();
	}

	if (!doc_class_name.is_empty()) {
		HashMap<String, DocData::ClassDoc>::ConstIterator F = doc_data->class_list.find(doc_class_name);
		if (!F) {
			doc_class_name = String();
		}
	}
```

Notice how there are two cases that set the value of `doc_class_name` and only after that we do the lookup.

Your first attempt checked exclusively for one case of the two here, and your second is checking exclusively for the other case. I think we should still check for both:

```cpp

	doc_class_name = script_base->get_global_name();
	if (doc_class_name.is_empty()) {
		doc_class_name = script_base->get_path().trim_prefix("res://").quote();
	}

	if (!doc_class_name.is_empty() && !doc_data->class_list.find(doc_class_name)) {
		doc_class_name = String();
	}
```